### PR TITLE
PQ-2 Create Models and Liquibase changesets to create patient_queueing table

### PR DESCRIPTION
**Pull request by slubwama**
Create Models and Liquibase changesets to create patient_queueing table
https://issues.openmrs.org/browse/PQ-2
This Replaces #3 whose changes Failed to squash into one commit

### CODE REVIEW

**Comment by slubwama** [comment]
@dkayiwa I opened #4 one and closed #3 due to failure  in squashing

File: api/src/main/java/org/openmrs/module/patientqueueing/PatientQueueingActivator.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 public class PatientQueueingActivator extends BaseModuleActivator {
 	
-	private Log log = LogFactory.getLog(this.getClass());
+	private Log log = LogFactory.getLog(PatientQueueingActivator.class);
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/PatientQueueingActivator.java]: This should be static final as per https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-loggingLogging

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    private Encounter encounter;
+
+    @Column(name = "status", length = 255)
+    private String status;
+
+    @Column(name = "queue_number", length = 255)
+    private String queueNumber;
+
+    @Column(name = "priority")
+    private Integer priority;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:55]: What values will priority have?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    private Location locationTo;
+
+    @ManyToOne
+    @JoinColumn(name = "encounter_id")
+    private Encounter encounter;
+
+    @Column(name = "status", length = 255)
+    private String status;
+
+    @Column(name = "queue_number", length = 255)
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:51]: What is queue number?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    @ManyToOne
+    @JoinColumn(name = "location_to")
+    private Location locationTo;
+
+    @ManyToOne
+    @JoinColumn(name = "encounter_id")
+    private Encounter encounter;
+
+    @Column(name = "status", length = 255)
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:48]: What will be contained in status?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    @GeneratedValue
+    @Column(name = "patient_queue_id")
+    private Integer patientQueueId;
+
+    @ManyToOne
+    @JoinColumn(name = "patient_id")
+    private Patient patient;
+
+    @ManyToOne
+    @JoinColumn(name = "provider_id")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:33]: Is the the provider that the patient is queued to see?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    private Provider provider;
+
+    @ManyToOne
+    @JoinColumn(name = "location_from")
+    private Location locationFrom;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:38]: What is location from and to?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+    @ManyToOne
+    @JoinColumn(name = "location_to")
+    private Location locationTo;
+
+    @ManyToOne
+    @JoinColumn(name = "encounter_id")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:45]: What is the use of encounter?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 5 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:33]: Yes

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 6 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:38]: Its Important to know where the patient started the queue from and also the location to where they are going

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 7 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:45]: This is to Help reference data on a queue. for example if a queue is generated by submission of an encounter some information on orders my be needed in the next location. an encounter is good reference of such data

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 4 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:48]: For example Completed, Rescheduled, Postponed. etc

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 3 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:51]: Number assigned to a patient on a given day at a health center

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 2 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:55]: This is a number assigned to the patient basing on urgency towards a review or treatment. For example Should be able to weight patients based on urgency.

**Comment by slubwama** [comment]
@dkayiwa Responded to all Questions and reacted to all changes nneded

**Comment by slubwama** [comment]
@dkayiwa ETA on this PR

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 4 above)
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:48]: Shouldn't this be an enumeration instead of an open ended string?

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:48]: Thinking about this again, how different is this from the appointmentscheduling module?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 3 above)
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:51]: What is this used for?

File: api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java (hunk 2 above)
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java:55]: Is this a real or hypothetical requirement?

**Pull request merged**